eth-providers/src/base-provider.ts

```diff
@@ -1606,13 +1632,27 @@ export abstract class BaseProvider extends AbstractProvider {
     return this.getTransactionReceiptAtBlock(txHash, targetBlockHash.toHex());
   };
 
-  _isTXPending = async (txHash: string): Promise<boolean> => {
+  _getPendingTX = async (txHash: string): Promise<TX | null> => {
     const pendingExtrinsics = await this.api.rpc.author.pendingExtrinsics();
-    for (const e of pendingExtrinsics) {
-      if (e.hash.toHex() === txHash) return true;
-    }
+    const targetExtrinsic = pendingExtrinsics.find(e => e.hash.toHex() === txHash);
+
+    if (!targetExtrinsic) return null;
 
-    return false;
+    const args = (targetExtrinsic.method.toJSON() as ExtrinsicMethodJSON).args;
+
+    return {
+      from: await this.getEvmAddress(targetExtrinsic.signer.toString()),
+      to: args.action.Call ? args.action.Call : null,
+      blockHash: null,
+      blockNumber: null,
+      transactionIndex: null,
+      hash: txHash,
+      nonce: targetExtrinsic.nonce.toNumber(),
+      value: args.value,
+      gasPrice: args.storage_limit,   // TODO: reverse calculate if needed
+      gas: args.gas_limit,
+      input: args.input,
+    }
```

Comment:
missed
```
      v: DUMMY_V,
      r: DUMMY_R,
      s: DUMMY_S,
```

Reply:
And check `extrinsic.method.section.toUpperCase() === "EVM"`

Reply:
that's a good point, I will add it tomorrow.